tests/test_pieces.py

```diff
@@ -98,4 +98,236 @@ def test_black_pawn_cannot_move_down_two_squares_if_already_moved():
+
+        # Act
+        moves = pawn.get_available_moves(board)
+
+        # Assert
+        assert obstructing_square not in moves
+
+    @staticmethod
+    def test_black_pawn_cannot_move_two_squares_if_piece_two_in_front():
+
+        # Arrange
+        board = Board.empty()
+        pawn = Pawn(Player.BLACK)
+        pawn_square = Square.at(6, 4)
+        board.set_piece(pawn_square, pawn)
+
+        obstructing_square = Square.at(4, 4)
+        obstruction = Pawn(Player.WHITE)
+        board.set_piece(obstructing_square, obstruction)
+
+        # Act
+        moves = pawn.get_available_moves(board)
+
+        # Assert
+        assert obstructing_square not in moves
+
+    @staticmethod
+    def test_white_pawn_cannot_move_two_squares_if_piece_one_in_front():
+
+        # Arrange
+        board = Board.empty()
+        pawn = Pawn(Player.WHITE)
+        pawn_square = Square.at(1, 4)
+        board.set_piece(pawn_square, pawn)
+
+        obstructing_square = Square.at(2, 4)
+        obstruction = Pawn(Player.BLACK)
+        board.set_piece(obstructing_square, obstruction)
+
+        # Act
+        moves = pawn.get_available_moves(board)
+
+        # Assert
+        assert Square.at(3, 4) not in moves
+
+    @staticmethod
+    def test_black_pawn_cannot_move_two_squares_if_piece_one_in_front():
+
+        # Arrange
+        board = Board.empty()
+        pawn = Pawn(Player.BLACK)
+        pawn_square = Square.at(6, 4)
+        board.set_piece(pawn_square, pawn)
+
+        obstructing_square = Square.at(5, 4)
+        obstruction = Pawn(Player.WHITE)
+        board.set_piece(obstructing_square, obstruction)
+
+        # Act
+        moves = pawn.get_available_moves(board)
+
+        # Assert
+        assert Square.at(4, 4) not in moves
+
+    @staticmethod
+    def test_white_pawn_cannot_move_at_top_of_board():
+
+        # Arrange
+        board = Board.empty()
+        pawn = Pawn(Player.WHITE)
+        square = Square.at(7, 4)
+        board.set_piece(square, pawn)
+
+        # Act
+        moves = pawn.get_available_moves(board)
+
+        # Assert
+        assert len(moves) == 0
+
+    @staticmethod
+    def test_black_pawn_cannot_move_at_bottom_of_board():
+
+        # Arrange
+        board = Board.empty()
+        pawn = Pawn(Player.BLACK)
+        square = Square.at(0, 4)
+        board.set_piece(square, pawn)
+
+        # Act
+        moves = pawn.get_available_moves(board)
+
+        # Assert
+        assert len(moves) == 0
+
+    @staticmethod
+    def test_white_pawns_can_capture_diagonally():
+
+        # Arrange
+        board = Board.empty()
+        pawn = Pawn(Player.WHITE)
+        pawn_square = Square.at(3, 4)
+        board.set_piece(pawn_square, pawn)
+
+        enemy1 = Pawn(Player.BLACK)
+        enemy1_square = Square.at(4, 5)
+        board.set_piece(enemy1_square, enemy1)
+
+        enemy2 = Pawn(Player.BLACK)
+        enemy2_square = Square.at(4, 3)
+        board.set_piece(enemy2_square, enemy2)
```

Comment:
`enemy1` and `enemy2` can be renamed into something a bit more decriptive, maybe something indicating `left` or `right` diagonal?